modules/programs/awscli.nix

```diff
@@ -2,8 +2,17 @@
 
 let
   cfg = config.programs.awscli;
+
   iniFormat = pkgs.formats.ini { };
 
+  settingsPath =
+    if cfg.settingsPath != ""
+    then cfg.settingsPath
+    else "${config.home.homeDirectory}/.aws/config";
+  credentialsPath =
+    if cfg.credentialsPath != ""
+    then cfg.credentialsPath
+    else "${config.home.homeDirectory}/.aws/credentials";
```

Comment:
Move to a let block above config or change to using a `cfg.settingsPath = lib.mkOptionDefault "${config.home.homeDirectory}/.aws/config"` to simplify logic and just make it default, as intended.